### ISSUE TITLE
Conditional Configspace KeyError

### ISSUE THREAD

**Issue by JohannesHog** [automl/SMAC3#1058]
#### Description
I wanted to run an optimization with a conditional config space but got a Key Error.

#### Steps/Code to Reproduce
```
import random

from ConfigSpace import Categorical, ConfigurationSpace, EqualsCondition, UniformIntegerHyperparameter
from smac import Scenario, HyperparameterOptimizationFacade


def train_function(config, seed):
    return random.random()


if __name__ == '__main__':
    cs = ConfigurationSpace()
    condition_picker = Categorical(name="optimizer", items=['hp1', 'hp2'])
    conditional_hp1 = UniformIntegerHyperparameter(name="hp1", lower=1, upper=100, default_value=4, log=False)
    conditional_hp2 = UniformIntegerHyperparameter(name="hp2", lower=1, upper=100, default_value=4, log=False)
    cs.add_hyperparameters([condition_picker, conditional_hp1, conditional_hp2])
    use_hp1 = EqualsCondition(conditional_hp1, condition_picker, 'hp1')
    use_hp2 = EqualsCondition(conditional_hp2, condition_picker, 'hp2')
    cs.add_conditions([use_hp1, use_hp2])

    scenario = Scenario(configspace=cs,
                        output_directory='./smac_output',
                        walltime_limit=60,
                        n_workers=1,
                        )

    smac = HyperparameterOptimizationFacade(scenario=scenario, target_function=train_function)
    smac.optimize()
```

#### Expected Results
Code works and returns a random config as incumbent.

#### Actual Results
```
Traceback (most recent call last):
  File "/home/johannes/PycharmProjects/smac_bug/main.py", line 28, in <module>
    smac.optimize()
  File "/home/johannes/anaconda3/envs/smac_bug/lib/python3.10/site-packages/smac/facade/abstract_facade.py", line 314, in optimize
    incumbents = self._optimizer.optimize(data_to_scatter=data_to_scatter)
  File "/home/johannes/anaconda3/envs/smac_bug/lib/python3.10/site-packages/smac/main/smbo.py", line 309, in optimize
    self._add_results()
  File "/home/johannes/anaconda3/envs/smac_bug/lib/python3.10/site-packages/smac/main/smbo.py", line 430, in _add_results
    self.tell(trial_info, trial_value)
  File "/home/johannes/anaconda3/envs/smac_bug/lib/python3.10/site-packages/smac/main/smbo.py", line 219, in tell
    response = callback.on_tell_end(self, info, value)
  File "/home/johannes/anaconda3/envs/smac_bug/lib/python3.10/site-packages/smac/intensifier/abstract_intensifier.py", line 452, in on_tell_end
    self.intensifier.update_incumbents(info.config)
  File "/home/johannes/anaconda3/envs/smac_bug/lib/python3.10/site-packages/smac/intensifier/abstract_intensifier.py", line 594, in update_incumbents
    print_config_changes(rh.get_config(removed_incumbent_id), config, logger=logger)
  File "/home/johannes/anaconda3/envs/smac_bug/lib/python3.10/site-packages/smac/utils/configspace.py", line 172, in print_config_changes
    params = sorted([(param, incumbent[param], challenger[param]) for param in challenger.keys()])
  File "/home/johannes/anaconda3/envs/smac_bug/lib/python3.10/site-packages/smac/utils/configspace.py", line 172, in <listcomp>
    params = sorted([(param, incumbent[param], challenger[param]) for param in challenger.keys()])
  File "/home/johannes/anaconda3/envs/smac_bug/lib/python3.10/site-packages/ConfigSpace/configuration.py", line 191, in __getitem__
    raise KeyError(key)
KeyError: 'hp2'

Process finished with exit code 1
```

#### Versions
This happened with smac version 2.0.1

**Comment by eddiebergman** [comment]
Hiyo, this was fixed in #1043 with other updates done in the unmerged #1051. Not sure when a release will next be done but you can pull and use the development version which includes the fix of #1043

**Comment by JohannesHog** [comment]
Thanks, that helps!
